beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -964,7 +988,8 @@ impl<T: BeaconChainTypes> OverflowLRUCache<T> {
         let has_missing_columns = pending_components.verified_data_columns.len() < num_of_columns;
 
         has_missing_columns
-            && !pending_components.reconstruction_started
+            // for simplicity now, we only reconstruct columns for one block at a time.
+            && self.reconstructing_block_root.lock().is_none()
```

Comment:
This silent skip is dangerous. We want to be intentional on when we NOT reconstruct and why. If the network experiences two data incomplete blocks close together, all supernodes could simultaneously hit this condition and not reconstruct.

If we really want to implement this "only 1 reconstruction at once" the right place to impose work limitation is the beacon processor. Here it should always go ahead (my preferred), wait (I won't recommend) or through an error (also bad).

Reply:
Yeah good point - I didn't realized that when I made the change. Maybe we can stick that field back to the `PendingComponents`

Reply:
Done